<?xml version="1.0"?>
<launch>

  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <arg name="robot_ip"/>

  <arg name="prefix" default="" />  
  <arg name="max_velocity" default="10.0"/> <!-- [rad/s] -->
  <!-- robot model -->
  <include file="$(find aubo_description)/launch/aubo_i5_upload.launch" />



  <!-- Load hardware interface -->
  <node name="aubo_hardware_interface" pkg="aubo_new_driver" type="aubo_new_driver" output="screen">
    <param name="robot_ip_address" type="str" value="$(arg robot_ip)"/>
    <param name="use_ros_control" type="bool" value="true"/>
  </node>

  <!-- Load controller settings -->
  <rosparam file="$(find aubo_new_driver)/config/aubo_i5_controllers.yaml" command="load"/>

  <!-- spawn controller manager -->
  <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="joint_state_controller pos_based_pos_traj_controller" />

  <!-- load other controller --> 
  <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
    output="screen" args="load vel_based_pos_traj_controller" /> 

  <!-- Convert joint states to /tf tranforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

</launch>

